src/sklearn_raster/features.py

```diff
@@ -8,59 +8,59 @@
 import xarray as xr
 from numpy.typing import NDArray
 
-from .types import ArrayUfunc, ImageType, NoDataType
+from .types import ArrayUfunc, FeatureType, NoDataType
 from .ufunc import UfuncSampleProcessor
-from .utils.image import image_to_samples
+from .utils.features import reshape_to_samples
 from .utils.wrapper import map_method_over_tuples
 
 
-class Image(Generic[ImageType], ABC):
-    """A wrapper around a multi-band image"""
+class FeatureArray(Generic[FeatureType], ABC):
+    """A wrapper around an n-dimensional array of features."""
 
-    band_dim_name: str | None = None
-    band_dim: int = 0
-    band_names: NDArray
+    feature_dim_name: str | None = None
+    feature_dim: int = 0
+    feature_names: NDArray
 
-    def __init__(self, image: ImageType, nodata_input: NoDataType = None):
-        self.image = image
-        self.n_bands = self.image.shape[self.band_dim]
+    def __init__(self, features: FeatureType, nodata_input: NoDataType = None):
+        self.features = features
```

Comment:
I think this plays into my earlier comment of `FeatureType` to `FeatureArrayType`.  I may be not understanding, but I view the data being passed into this `__init__` as a *singular* array of features, whereas `features` to me seems like a collection.  Would `feature_arr` or `feature_array` work here?  And if I'm off base, please ignore.